packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2133,6 +2122,173 @@ export class MergeTree {
         return segRef;
     }
 
+    private ensureLocalPartialsComputed(refSeq: number): void {
+        if (!this.localPartialsComputed) {
+            const rebaseCollabWindow = new CollaborationWindow();
+            rebaseCollabWindow.loadFrom(this.collabWindow);
+            if (refSeq < this.collabWindow.minSeq) {
+                rebaseCollabWindow.minSeq = refSeq;
+            }
+            this.root.partialLengths = PartialSequenceLengths.combine(
+                this.root,
+                rebaseCollabWindow,
+                true,
+                true,
+            );
+            this.localPartialsComputed = true;
+        }
+    }
+
+    // Segments should either be removed remotely, removed locally, or inserted locally
+    private normalizeAdjacentSegments(affectedSegments: List<ISegment>): void {
+        const currentOrderTest = Array.from(affectedSegments.map(({ data }) => data));
+        // Eagerly demand this since we're about to shift elements in the list around
```

Comment:
seems like these two lines could be combined and reduce an array allocation and enumeration

Reply:
oops yeah I meant to fix this, good catch. Using the second param on Array.from() should reduce to only one array allocation..